include/nlohmann/detail/input/input_adapters.hpp

```diff
@@ -269,20 +280,20 @@ struct wide_string_input_helper<WideStringType, 2>
     }
 };
 
-template<typename WideStringType>
+// Wraps another input apdater to convert wide character types into individual bytes.
+template<typename BaseInputAdapter, typename WideCharType>
 class wide_string_input_adapter
 {
```

Comment:
I am thinking if we could use more general name, like `wide_char_xxx` or `wide_byte_xxx`?
PS: never mind. Basically, I think all your changes are amazing!